yt_dlp/postprocessor/ffmpeg.py

```diff
@@ -1173,3 +1186,160 @@ def run(self, info):
             'ext': ie_copy['ext'],
         }]
         return files_to_delete, info
+
+
+class FFmpegProgressTracker:
+
+    def __init__(self, info_dict: dict, ffmpeg_args: List, ffmpeg_proc: Popen, hook_progress: Callable, running_status_value: str, action_bytes_key: str):
+        self.info_dict = info_dict
+        self.ffmpeg_args = ffmpeg_args
+        self.ffmpeg_proc = ffmpeg_proc
+        self.hook_progress = hook_progress
+        self.running_status_value = running_status_value
+        self.action_bytes_key = action_bytes_key
+
+    def ffmpeg_track_progress(self) -> Tuple[str, str, int]:
+        # Get ffmpeg progress by capturing and parsing the output with the '-progress' option
+        duration_to_track, duration = self.compute_duration_to_track()
+        total_filesize = self.compute_total_filesize(duration_to_track, duration)
+        started = time.time()
+        status = {
+            'filename': self.info_dict.get('_filename'),
+            'status': self.running_status_value,
+            'total_bytes': total_filesize,
+            'elapsed': time.time() - started
+        }
+        # Parse ffmpeg_output
+        progress_pattern = re.compile(
+            r'(frame=\s*(?P<frame>\S+)\nfps=\s*(?P<fps>\S+)\nstream_0_0_q=\s*(?P<stream_0_0_q>\S+)\n)?bitrate=\s*(?P<bitrate>\S+)\ntotal_size=\s*(?P<total_size>\S+)\nout_time_us=\s*(?P<out_time_us>\S+)\nout_time_ms=\s*(?P<out_time_ms>\S+)\nout_time=\s*(?P<out_time>\S+)\ndup_frames=\s*(?P<dup_frames>\S+)\ndrop_frames=\s*(?P<drop_frames>\S+)\nspeed=\s*(?P<speed>\S+)\nprogress=\s*(?P<progress>\S+)')
+        retval = self.ffmpeg_proc.poll()
+        ffmpeg_stdout_buffer = ''
+        ffmpeg_stderr_buffer = ''
+        while retval is None:
+            ffmpeg_stdout = self.ffmpeg_proc.stdout.readline() if self.ffmpeg_proc.stdout else ''
+            ffmpeg_stderr_buffer = self.ffmpeg_proc.stderr.readline() if self.ffmpeg_proc.stderr else ''
+            if ffmpeg_stderr_buffer != '':
+                sys.stderr.write(ffmpeg_stderr_buffer)
+                ffmpeg_stderr_buffer = ''
+            if ffmpeg_stdout == '':
+                continue
+            ffmpeg_stdout_buffer += ffmpeg_stdout
+            ffmpeg_prog_infos = re.match(progress_pattern, ffmpeg_stdout_buffer)
+            if not ffmpeg_prog_infos:
+                continue
+            sys.stdout.write(ffmpeg_stdout_buffer)
```

Comment:
wouldn't it break `-o -`?

Reply:
I'm sorry I'll need more context because I don't understand your comment

Reply:
~~yt-dlp has an option to write downloaded data to stdout (`-o -`) blablabla~~
Ah when it has `-o -` it should work ig